src/extra/gfx/material_utils.hpp

```diff
@@ -76,15 +96,46 @@ inline void initConstantShaderParams(MaterialParameters &out, SHTable &paramsTab
   }
 }
 
-inline void initReferencedShaderParams(SHContext *shContext, SHShaderParameters &shShaderParameters, SHTable &paramsTable) {
+inline void initConstantTextureParams(MaterialParameters &out, SHTable &texturesTable) {
   SHTableIterator it{};
   SHString key{};
   SHVar value{};
-  paramsTable.api->tableGetIterator(paramsTable, &it);
-  while (paramsTable.api->tableNext(paramsTable, &it, &key, &value)) {
+  texturesTable.api->tableGetIterator(texturesTable, &it);
+  while (texturesTable.api->tableNext(texturesTable, &it, &key, &value)) {
+    TexturePtr texture;
+    varToTexture(value, texture);
+    out.set(key, texture);
+  }
+}
+
+inline void initReferencedShaderParams(SHContext *shContext, std::vector<SHBasicShaderParameter> &outParams, SHTable &inTable) {

```

Comment:
I would have the *out* parameter always last (or always first) if possible for consistency across multiple functions.